packages/database/src/core/SyncTree.ts

```diff
@@ -423,6 +423,38 @@ export function syncTreeRemoveEventRegistration(
   return cancelEvents;
 }
 
+/**
+ * This function was added to support non-listener queries,
+ * specifically for use in repoGetValue. It sets up all the same
+ * local cache data-structures (SyncPoint + View) that are
+ * needed for listeners without installing an event registration.
+ * If `query` is not `loadsAllData`, it will also provision a tag for
+ * the query so that query results can be merged into the sync
+ * tree using existing logic for tagged listener queries.
+ *
+ * @param syncTree - Synctree to add the query to.
+ * @param query - Query to register
+ * @returns tag as a string if query is not a default query, null if query is not.
+ */
+export function syncTreeRegisterQuery(syncTree: SyncTree, query: QueryContext) {
```

Comment:
The asymmetry of `syncTreeRegisterQuery` without `syncTreeUnregisterQuery` might raise some eyebrows. I would either add:

```
export function syncTreeUnregisterQuery(syncTree: SyncTree, query: QueryContext) {
  const cancels = syncTreeRemoveEventRegistration(
    repo.serverSyncTree_,
    query,
    null
   );
   if (cancels.length > 0) {
     repoLog(repo, 'ignoring unexpected cancel events in get for query ' + stringify(query));
   }
}
```

or note the reason we don't have `syncTreeUnregisterQuery` in the comments. I'm fine with either.

Reply:
I think this additional function makes sense

Reply:
Ended up just keeping it in `repoGetValue` since we aren't logging in `SyncTree` and it didn't seem like the right place to put the function.